### ISSUE TITLE
C struct definitions with no trailing {

### ISSUE THREAD

**Issue by jaytiar** [abusalimov/SublimeCImproved#24]
Great package - much improved over the basic one. Thanks!
I found a small issue - we use C coding style that requires structure types to be defined like this:
`typedef struct name_of_struct`
`{`
`  fields;`
`} NAME_OF_STRUCT;`

The syntax picks up the NAME_OF_STRUCT correctly as a typedef type, but does not pick up the lower case structure name name_of_struct as an entity.name.class.c scope, which means it doesn't appear in the symbol index.

I think the problem is that the relevant capture assumes the `{` comes on the end of the line the `struct` appears on.

I think the fix is to allow this variant in the capture. So at line 251 of C Improved.tmLanguage, we end up with this:
`<key>match</key> <string>\b([A-Za-z_]\w*+)(?=(?:\s|/\*.*?\*/)*+(?:\{|$))</string>`

instead of the old version:
`<key>match</key> <string>\b([A-Za-z_]\w*+)(?=(?:\s|/\*.*?\*/)*+\{)</string>`

If you agree, could you make the fix? 
Thanks!

**Comment by abusalimov** [comment]
Thanks for reporting!
Yep, I know about this bug. The problem is that there's no way to distinguish the valid case ...

```c
// struct definition, should appear in index
struct name_of_struct
{
    ...
}
```

... from this one:
```c
// variable definition of that type, shouldn't be indexed
struct name_of_struct
variable;
```

The fix would make both of these cases be recognized as `entity.name`, including that odd variable definition.

On the other hand, I've never seen anyone declaring variables like that, and the proposed fix probably covers 95% of possible use cases, recognizing them properly. So I'll likely add this fix once I have some spare time.

**Comment by jon-rowland** [comment]
Thanks for the speedy response!
Yes, I agree it's hard to distinguish those cases, but I've also never seen anyone declare variables like that, whereas the definition across two lines is much more common, so I agree making the fix will cover the majority of cases.

**Comment by abusalimov** [comment]
Released in v1.8.0 and should reach the Package Control soon. Thanks again for the suggestion!
